dev/make-distribution.sh

```diff
@@ -71,6 +72,9 @@ while (( "$#" )); do
     --pip)
       MAKE_PIP=true
       ;;
+    --r)
+      MAKE_R=true
```

Comment:
FWIW if you want this to get picked up by the official release building procedure we also need to edit release-build.sh [1]. Can you coordinate this with @rxin ?

[1] https://github.com/apache/spark/blob/d9eb4c7215f26dd05527c0b9980af35087ab9d64/dev/create-release/release-build.sh#L220